### PR TITLE
[release] 0.2.121

### PR DESCRIPTION
**Pull request by danking**
I also had to fix issues that pyright discovered.

### CODE REVIEW

**Review by patrick-schultz — APPROVED** [review]

**Comment by danking** [comment]
I'm gonna run test-dataproc-37 and -38 before I merge.

**Comment by danking** [comment]
https://ci.hail.is/batches/7917419

**Comment by danking** [comment]
Dataproc tests passed, woohoo!

I'm gonna hold off on merging until some bug fixes land:
1. https://github.com/hail-is/hail/pull/13505
2. https://github.com/hail-is/hail/pull/13500

**Pull request merged**
